document/core/exec/runtime.rst

```diff
@@ -608,6 +660,66 @@ This definition allows to index active labels surrounding a :ref:`branch <syntax
    The selected label is identified through the :ref:`label index <syntax-labelidx>` :math:`l`, which corresponds to the number of surrounding |LABEL| instructions that must be hopped over -- which is exactly the count encoded in the index of a block context.
 
 
+.. index:: ! throw context, event, exception, throw address, catch block
+.. _syntax-ctxt-throw:
+
+Throw Contexts
+..............
+
+In order to specify the reduction of :ref:`try-catch blocks <syntax-try>`
+with the help of the administrative instructions |THROWADDR|, |EXNREFADDR|, and |CATCHN|,
+the following syntax of *throw contexts* is defined, as well as associated structural rules:
+
+.. math::
+   \begin{array}{llll}
+   \production{(throw contexts)} & \XT &::=&
+     \val^\ast~[\_]~\instr^\ast \\ &&|&
+     \LABEL_n\{\instr^\ast\}~\XT~\END \\ &&|&
+     \CATCHN_n\{\instr^\ast\}~\XT~\END \\ &&|&
+     \FRAME_n\{F\}~\XT~\END \\
+   \end{array}
+
+.. math::
+   \begin{array}{rcl}
+   S;~F;~\CATCHN_m~\{\instr^\ast\}~\val^m~\END &\stepto& S;~F;~\val^m \\
+   S;~F;~\CATCHN_m~\{\instr^\ast\}~\XT[\val^n~[\_]~(\THROWADDR~a)]~\END &\stepto&
+      S;~F;~\LABEL_m~\{\}~(\EXNREFADDR~a~\val^n)~{\instr}^\ast~\END \\
+   && \hspace{-12ex} (\iff S.\SEVENTS[a]=\{\EVATTRIBUTE~\EXCEPTION, \EVTYPE~[t^n]\to[]\}) \\
+   \end{array}
+
+Throw contexts allow to potentially capture the throw of an exception event by surrounding catch blocks.
+
+.. note::
+   For example, catching a simple :ref:`throw <exec-throw>` in a :ref:`try-catch block <exec-try>` would be as follows.
+
+   Let :math:`x` be the :ref:`event index <syntax-eventidx>` of an event in :math:`F.\AMODULE`, such that
+
+   .. math::
+      F.\AMODULE.\MIEVENTS[x] = \{ \EVIATTRIBUTE~\EXCEPTION, \EVITYPE~[t^n]\to[]\} = S.\SEVENTS[a]
+
+   Let :math:`\expand_F(\X{bt}) = [t^n] \to [\EXNREF]`, and consider arbitrary :ref:`values <syntax-val>` :math:`(val : t)^n`.
```

Comment:
This seems tangential to the example, maybe remove?

Reply:
I added the bt expansion to emphasize that this instruction will return the exception reference as a value, not throw it. Then I guess I got carried away and added all the details.  Since my initial point is also spelled out in the end of the example, I will remove these lines as you suggest.

Reply:
Done.